src/dmd/backend/cg.d

```diff
@@ -49,111 +51,112 @@ Symbol *tls_get_addr_sym;       // function __tls_get_addr
+                   FLstack,FLregsave,FLfuncarg,
+                   FLndp, FLfardata,
+                 ])
+    {
+        datafl[fl] = true;
+    }
+    return datafl;
+} ();
+
+
+/// Is fl on the stack?
+bool[FLMAX] stackfl =
+() {
+    bool[FLMAX] stackfl;
+    foreach (fl; [ FLauto,FLfast,FLpara,FLcs,FLfltreg,FLallocatmp,FLbprel,FLstack,FLregsave,
+                   FLfuncarg,
+                   FLndp,
+                 ])
+    {
+        stackfl[fl] = true;
+    }
+    return stackfl;
+} ();
+
+/// What segment register is associated with it?
+ubyte[FLMAX] segfl =
+() {
+    ubyte[FLMAX] segfl;
+
+    // Segment registers
+    enum ES = 0;
+    enum CS = 1;
+    enum SS = 2;
+    enum DS = 3;
+    enum NO = ubyte.max;        // no register
+
+    foreach (fl, ref seg; segfl)
+    {
+        switch (fl)
         {
-            datafl[fl] = 1;
+            case 0:              seg = NO;  break;
+            case FLconst:        seg = NO;  break;
+            case FLoper:         seg = NO;  break;
+            case FLfunc:         seg = CS;  break;
+            case FLdata:         seg = DS;  break;
+            case FLudata:        seg = DS;  break;
+            case FLreg:          seg = NO;  break;
+            case FLpseudo:       seg = NO;  break;
+            case FLauto:         seg = SS;  break;
+            case FLfast:         seg = SS;  break;
+            case FLstack:        seg = SS;  break;
+            case FLbprel:        seg = SS;  break;
+            case FLpara:         seg = SS;  break;
+            case FLextern:       seg = DS;  break;
+            case FLcode:         seg = CS;  break;
+            case FLblock:        seg = CS;  break;
+            case FLblockoff:     seg = CS;  break;
+            case FLcs:           seg = SS;  break;
+            case FLregsave:      seg = SS;  break;
+            case FLndp:          seg = SS;  break;
+            case FLswitch:       seg = NO;  break;
+            case FLfltreg:       seg = SS;  break;
+            case FLoffset:       seg = NO;  break;
+            case FLfardata:      seg = NO;  break;
+            case FLcsdata:       seg = CS;  break;
+            case FLdatseg:       seg = DS;  break;
+            case FLctor:         seg = NO;  break;
+            case FLdtor:         seg = NO;  break;
+            case FLdsymbol:      seg = NO;  break;
+            case FLgot:          seg = NO;  break;
+            case FLgotoff:       seg = NO;  break;
+            case FLtlsdata:      seg = NO;  break;
+            case FLlocalsize:    seg = NO;  break;
+            case FLframehandler: seg = NO;  break;
+            case FLasm:          seg = NO;  break;
+            case FLallocatmp:    seg = SS;  break;
+            case FLfuncarg:      seg = SS;  break;
+
+            default:
+                assert(0);
         }
-        return datafl;
-    } ();
-
+    }
+
+    return segfl;
+} ();
+
+/// Is fl in the symbol table?
+bool[FLMAX] flinsymtab =
```

Comment:
`immutable`?

Reply:
immutable would require more refactoring, as it is not C++ compatible. So not this time.

Reply:
What needs to be compatible with C++ ? Is it for DMD, or DMC ?

Reply:
Also this is removing `C++` linkage from those data structure so they are not "C++ compatible" anymore

Reply:
no, it isn't. Look at the whole file.

Reply:
> What needs to be compatible with C++ ?

D mangles the name to include the module name. C++ does not. Therefore, switching to D linkage means the modules need to be reorganized. This is not the PR to do that.